### PR TITLE
Networking dashboard and discovery tool refactor

### PR DESCRIPTION
**Pull request by ludoo**
@juliocc this is now ready for initial review. I will promote from draft after we have checked some of the more intricate logic with  @aurelienlegrand .

### CODE REVIEW

**Review by juliocc — APPROVED** [review]
great work

File: blueprints/cloud-operations/network-dashboard/src/custom-quotas.sample [hunk 1]
```diff
@@ -0,0 +1,8 @@
+projects:
+  tf-playground-svpc-net:
+    global:
+      INTERNAL_FORWARDING_RULES_PER_NETWORK: 750
+networks:
+  # TODO: what are the quotas that can be overridden at the network level?
```
[the remark this answers is not in the document]
Comment (ludoo) [reply blueprints/cloud-operations/network-dashboard/src/custom-quotas.sample:6]: I'm fine either way: if we add every quota the file becomes full of stuff users won't probably need, and which we will need to maintain. While if users ask for a quota raise, they usually know what the quota is. Up to you, add everything if you prefer it this way.

File: blueprints/cloud-operations/network-dashboard/deploy-cloud-function/README.md [hunk 2]
```diff
@@ -0,0 +1,78 @@
...
+
+```tfvars
+# cloud_function_config = {
+#   debug = true
+# }
+discovery_config = {
+  discovery_root     = "organizations/1234567890"
+  monitored_folders  = ["3456789012", "7890123456"]
+  monitored_projects = []
+  custom_quota_file  = "../src/custom-quotas.yaml"
```
[the remark this answers is not in the document]
Comment (ludoo) [reply blueprints/cloud-operations/network-dashboard/deploy-cloud-function/README.md:33]: no, we don't want users passing in the sample: it should the the name of the file users actually compile if they need it, let me add a comment

File: blueprints/cloud-operations/network-dashboard/deploy-cloud-function/main.tf [hunk 3]
```diff
@@ -0,0 +1,138 @@
...
+  time_zone = "UTC"
+
+  pubsub_target {
+    attributes = {}
+    topic_name = module.pubsub.topic.id
+    data = base64encode(jsonencode({
+      discovery_root = var.discovery_config.discovery_root
+      folders        = var.discovery_config.monitored_folders
+      projects       = var.discovery_config.monitored_projects
+      op_project     = module.project.project_id
```
[the remark this answers is not in the document]
Comment (aurelienlegrand) [reply blueprints/cloud-operations/network-dashboard/deploy-cloud-function/main.tf:100]: What do you think about "metrics_project"?

**Review by aurelienlegrand — APPROVED** [review]
LGTM, thanks Ludo for this amazing contribution and large scale refactoring!
I was able to test it and the code runs about 6 times faster than before your refactoring. Well done!

File: blueprints/cloud-operations/network-dashboard/deploy-cloud-function/main.tf (hunk 3 above)
[the remark this answers is not in the document]
Comment (ludoo) [reply blueprints/cloud-operations/network-dashboard/deploy-cloud-function/main.tf]: I renamed to monitoring_project, if you prefer metrics_project we can still change it.

File: blueprints/cloud-operations/network-dashboard/src/plugins/discover-cai-compute.py [hunk 4]
```diff
@@ -0,0 +1,232 @@
...
+function, which then delegates parsing of resource-level attributes to smaller
+specialized functions, one per resource type.
+'''
+
+import logging
+
+from . import HTTPRequest, Level, Resource, register_init, register_discovery
+from .utils import parse_cai_results
+
+# https://content-cloudasset.googleapis.com/v1/organizations/436789450919/assets?contentType=RESOURCE&assetTypes=compute.googleapis.com/Network
```
Comment (aurelienlegrand) [inline blueprints/cloud-operations/network-dashboard/src/plugins/discover-cai-compute.py:27]: @ludoo what is this comment for? Should we clean it up?

Comment (ludoo) [reply blueprints/cloud-operations/network-dashboard/src/plugins/discover-cai-compute.py:27]: yep we can remove it

**Pull request merged**